lib/resty/etcd/v3.lua

```diff
@@ -710,10 +714,12 @@ local function request_chunk(self, method, path, opts, timeout)
         if not body then
             return nil, "failed to decode json body: " .. (err or " unkwon")
         elseif body.error and body.error.http_code >= 500 then
-            if health_check.conf ~= nil then
+            if health_check.conf.shm_name ~= nil then
                 -- health_check retry should do nothing here
                 -- and let connection closed to create a new one
                 health_check.report_failure(endpoint.http_host)
+            else
+                health_check.report_round_robin_target_failure(endpoint.http_host)
```

Comment:
We could move all the codes related to the health check and can be moved to `health_check` to `health_check`